arch/arm/core/cortex_m/mpu/arm_mpu.c

```diff
@@ -52,6 +52,32 @@ static void _region_init(u32_t index, struct arm_mpu_region *region_conf)
 		index, region_conf->base, region_conf->attr);
 }
 
+/* ARM Core MPU Driver API Implementation for ARM MPU */
+
+/**
+ * @brief enable the MPU
+ */
+void arm_core_mpu_enable(void)
+{
+	/* Enable MPU and use the default memory map as a
+	 * background region for privileged software access.
+	 */
+	MPU->CTRL = MPU_CTRL_ENABLE_Msk | MPU_CTRL_PRIVDEFENA_Msk;
+
+	/* Make sure that all the registers are set before proceeding */
+	__DSB();
+	__ISB();
```

Comment:
ISB is not needed (unless access to the few following instructions is revoked and we should be sure that crash is imminent in such case).

Reply:
Well, this is the recommendation from ARM:
https://static.docs.arm.com/100699/0100/armv8m_architecture_memory_protection_unit_100699_0100_00_en.pdf?_ga=2.255697806.720744673.1531742502-979634574.1512129834

Also, in this way, we do not depend on which instructions are executed after enabling the MPU